src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -134,26 +157,50 @@ public String getSecret(String alias) {
         String secret = null;
         try {
 
-            final VaultConfig config = new VaultConfig()
+            VaultConfig config = new VaultConfig()
                     .address(address)
-                    .token(rootToken)
+                    .token(accessToken)
                     .engineVersion(engineVersion)
                     .build();
 
             Vault vault = new Vault(config);
             Logical logical = vault.logical();
-            if (StringUtils.isEmpty(namespace)) {
+            if (!StringUtils.isEmpty(namespace)) {
                 logical = logical.withNameSpace(namespace);
             }
             secret = logical.read(sb.toString()).getData()
                     .get(VALUE_PARAMETER);
 
             if (StringUtils.isEmpty(secret)) {
-                LOG.error("Cannot read the vault secret from the HashiCorp vault. " +
-                        "Check whether the VAULT_TOKEN is correct and the secret path is available: " + sb.toString());
+                LOG.info("Cannot read the vault secret from the HashiCorp vault. " +
```

Comment:
Shouldn't this be a warn log ?

Reply:
Corrected in the refactor commit 913abdc33b35c37b20e73ed5189933bf23648a05